src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/tools/ExplosiveTool.java

```diff
@@ -52,6 +52,8 @@ public ExplosiveTool(Category category, SlimefunItemStack item, RecipeType recip
     @Override
     public ToolUseHandler getItemHandler() {
         return (e, tool, fortune, drops) -> {
+            if (e.isCancelled()) return;
```

Comment:
This is completely redundant as the Listener which calls this method already checks if it is cancelled.
Also missing brackets.

Reply:
> This is completely redundant as the Listener which calls this method already checks if it is cancelled.
> Also missing brackets.

No it doesn't ![Screenshot_2020-10-11-02-04-16-647_com.github.android.jpg](https://user-images.githubusercontent.com/54643600/95666811-2299b300-0b66-11eb-84d4-f3f41701fd20.jpg)

Reply:
It only does a material check.

Reply:
**ignoreCancelled = true**
```java
@EventHandler(priority = EventPriority.MONITOR, ignoreCancelled = true)
    public void onBlockBreak(BlockBreakEvent e) {
```

Reply:
> **ignoreCancelled = true**
> ```java
> @EventHandler(priority = EventPriority.MONITOR, ignoreCancelled = true)
>     public void onBlockBreak(BlockBreakEvent e) {
> ```

Yes. So listener is fired no matter if the even is cancelled. If that worked like it ignored cancelled events, then why is there a cancelled check below?

Reply:
Ok just looked up, I was wrong.

Reply:
ignoreCancelled is very ambiguous